contrib/sirupsen/logrus/example_test.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package logrus
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/sirupsen/logrus"
+)
+
+func ExampleHook() {
+	// Setup logrus, do this once at the beginning of your program
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	logrus.AddHook(&DDContextLogHook{})
+
+	// Setting output to Stdout so output can be validated
+	logrus.SetOutput(os.Stdout)
+
+	span, sctx := tracer.StartSpanFromContext(context.Background(), "mySpan")
+
+	// Pass the current span context to the logger (Time is set for consistency in output here)
+	cLog := logrus.WithContext(sctx).WithTime(time.Date(2000, 1, 1, 1, 1, 1, 0, time.UTC))
+	// Log as desired using the context-aware logger
+	cLog.Info("Completed some work!")
+
+	span.Finish()
```

Comment:
Why is this not coupled with the start of this span and used as the conventional `defer span.Finish()` ? Also, please note that you forgot to start & stop the tracer, or was that on purpose?

Reply:
It was partially on purpose in an effort to try to keep the example small. Additionally it makes the trace_id and span_id always `0` which made matching the output a bit easier.

I could add starting / stopping the tracer it would just mean we need to manually set the trace_id and span_id so they're consistent for the `// Output:`

Thoughts?

Reply:
I think setting the IDs manually is not worth it. The two options I see are:
* Add the tracer Start/Stop and drop the `// Output:` stuff and simply write the example output in a comment
* Keep the `// Output:` line but add a comment at the top of the example that the tracer needs to be started.

I'll leave it up to you which one to chose